src-tauri/src/main.rs

```diff
@@ -72,79 +71,90 @@ async fn control_grabbing(
 
-    wooting_macro_backend::MacroBackend::generate_directories();
+    MacroBackend::generate_directories().context("unable to generate config directories")?;
 
     let backend = MacroBackend::default();
 
-    info!("Running the macro backend");
-
     #[cfg(any(target_os = "windows", target_os = "linux"))]
-    backend.init().await;
+    // The backend is run only on Windows and Linux, on macOS it won't work.
+    info!("Running the macro backend");
+    if let Err(e) = backend.init().await {
+        eprintln!("Initialization error: {}", e);
+    };
 
+    // Read the options from the config.
     let set_autolaunch: bool = backend.config.read().await.auto_start;
     let set_launch_minimized: bool = backend.config.read().await.minimize_at_launch;
 
-    // Begin the main event loop. This loop cannot run on another thread on MacOS.
+    // Set the window behaviors (add buttons to the right click tray menu).
     let quit = CustomMenuItem::new("quit".to_string(), "Quit");
     let hide_show = CustomMenuItem::new("hide_show".to_string(), "Hide");
-
     let tray_menu = SystemTrayMenu::new()
         .add_item(hide_show)
         .add_native_item(SystemTrayMenuItem::Separator)
         .add_item(quit);
-
     let system_tray = SystemTray::new().with_menu(tray_menu);
 
+    // Initialize the main application
     tauri::Builder::default()
-        // This is where you pass in your commands
+        // State management is initialized
         .manage(backend)
+        // This is where commands shared with frontend are passed
         .invoke_handler(tauri::generate_handler![
             get_macros,
             set_macros,
             get_config,
             set_config,
-            control_grabbing,
-            get_monitor_data
+            control_grabbing
         ])
         .setup(move |app| {
             let app_name = &app.package_info().name;
-            let current_exe = current_exe().unwrap();
-
-            let auto_start = auto_launch::AutoLaunchBuilder::new()
-                .set_app_name(app_name)
-                .set_app_path(current_exe.as_path().to_str().unwrap())
-                .set_use_launch_agent(true)
-                .build()
-                .unwrap();
-
-            match set_autolaunch {
-                true => auto_start.enable().unwrap(),
-                false => {
-                    if let Err(e) = auto_start.disable() {
-                        match e {
-                            auto_launch::Error::Io(err) => match err.kind() {
-                                std::io::ErrorKind::NotFound => {
-                                    info!("Autostart is already removed, finished checking.")
-                                }
-                                _ => error!("{}", err),
-                            },
-                            _ => error!("{}", e),
-                        }
-                    }
-                }
+            if let Ok(current_exe) = current_exe() {
```

Comment:
The current_exe isn't used for anything else, so you could just derive it inside of the init_autostart. I can't really see us doing autostart for any other exe, so no need to make the function generic

Reply:
Tweaked: [d8b50af](https://github.com/WootingKb/wooting-macros/pull/194/commits/d8b50af2e3b54c7bd09a268658b8a68645f2873e)